### PR TITLE
Update to otel 0.17.0

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

**Comment by trask** [comment]
Draft for now. #1505 needs to be merged first.

File: agent/agent-tooling/spotbugs.exclude.xml [hunk 1]
```diff
@@ -57,5 +57,49 @@
...
+        <Class name="~com\.microsoft\.applicationinsights\.agent\.internal\.wasbootstrap\.configuration\..*" />
+    </Match>
+    <Match>
+        <Bug pattern="UWF_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD" />
+        <!-- used from agent-tooling -->
+        <Class name="~com\.microsoft\.applicationinsights\.agent\.internal\.wasbootstrap\.configuration\..*" />
+    </Match>
+    <Match>
+        <Bug pattern="CRLF_INJECTION_LOGS" />
+        <Class name="com.microsoft.applicationinsights.agent.internal.wasbootstrap.MainEntryPoint" />
```
Comment (trask) [inline agent/agent-tooling/spotbugs.exclude.xml:92]: rename `wasbootstrap`

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/AiBootstrapPackageProvider.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+import io.opentelemetry.javaagent.spi.BootstrapPackagesProvider;
+
+public class AiBootstrapPackageProvider implements BootstrapPackagesProvider {
+
+    @Override
+    public List<String> getPackagePrefixes() {
+        // TODO is this needed?
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/AiBootstrapPackageProvider.java:12]: in this PR?

File: agent/instrumentation/build.gradle [hunk 3]
```diff
@@ -104,18 +105,15 @@ shadowJar {
     }
 
     dependencies {
-        // exclude bootstrap dependencies from shadowJar
+        // exclude wasbootstrap dependencies from shadowJar
```
Comment (trask) [inline agent/instrumentation/build.gradle:108]:
```suggestion
        // exclude bootstrap dependencies from shadowJar
```

File: agent/agent-tooling/spotbugs.exclude.xml (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/spotbugs.exclude.xml:92]: I will rename in separate PR, it's an internal package and so many changes in this PR already

File: agent/instrumentation/build.gradle (hunk 3 above)
Comment (trask) [reply agent/instrumentation/build.gradle]: done

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/AiBootstrapPackageProvider.java (hunk 2 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/AiBootstrapPackageProvider.java]: it is needed, removed TODO

**Review by heyams — APPROVED** [review]
👍

**Pull request merged**